### ISSUE TITLE
[BUG] When update vlan network with invalid CIDR and try to delete it, process stuck in `Deleting` and lost all network connection

### ISSUE THREAD

**Issue by TachunLin** [harvester/harvester#3615]
**Describe the bug**

When we edit the manual route of vlan network config to invalid CIDR 
And trying to delete the vlan network. 

* The progress stuck in `Deleting` 
![image](https://user-images.githubusercontent.com/29251855/223667536-dc914d6c-135c-4c4e-bb29-1b5512ed04fc.png)

* Then display `Network error` 
![image](https://user-images.githubusercontent.com/29251855/223668228-c0a7edb4-89de-4645-b94d-2452aa3ab668.png)

* Found we lost the VIP dashboard connection 
![image](https://user-images.githubusercontent.com/29251855/223669371-83d54bec-a9f4-4014-a0fd-047f67640044.png)

* And also failed to access every nodes' IP

**To Reproduce**
1. Create a cluster network
2. Create a network config and set `ens1f1` for the uplink NIC
3. Create a vlan network and set manual route
4. Provide some incorrect CIDR and cancel
5. Provide valid CIDR in the range and gateway
6. Wait the vlan network become Active
7. Update the config of vlan network
8. Set invalid CIDR and cancel 
9. Delete the vlan network
10. Check the delete status
11. Check the dashboard VIP connection

**Expected behavior**

Should be able to delete vlan network after update it. 
And won't lose VIP connection

**Support bundle**

**Environment**
 - Harvester ISO version: `master-544cb2b2` (03/08)
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): 4 nodes bare machines

**Additional context**
Unable to ping and access Harvester dashboard through VIP
Also I could not reach every node's IP
- 10.84.132.8
- 10.84.132.9
- 10.84.132.10
- 10.84.132.11

**Comment by TachunLin** [comment]
Checking the VGA console, it looks like the node IP disappear
![image](https://user-images.githubusercontent.com/29251855/223690876-a146e5f5-c65a-4ecc-ac03-2557a6b7c820.png)


The `ip a` network information 
![image](https://user-images.githubusercontent.com/29251855/223690247-9471f4c5-f974-49e1-a1d3-2f6e954ef7de.png)

**Comment by TachunLin** [comment]
After checking the existing environment with @yaocw2020 today. 
The VIP and node IP connectivity have been recovered. It probably restored by operating system. 

Further checking the harvester-node-controller logs 
[harvester-network-controller-4kb9h.log](https://github.com/harvester/harvester/files/10927427/harvester-network-controller-4kb9h.log)

And checking the IP route table
```
harvester-268pz:/home/rancher # ip r
default via 10.84.135.254 dev mgmt-br proto dhcp 
10.52.0.4 dev cali0dfe14687b2 scope link 
10.52.0.6 dev caliefcd7e1ee1b scope link 
10.52.0.7 dev calib631ce3387e scope link 
10.52.0.12 dev cali04de3ecb9f4 scope link 
10.52.0.13 dev califf1ae95cf65 scope link 
10.52.0.16 dev cali51a22b382fe scope link 
10.52.0.19 dev cali2265c463b8f scope link 
10.52.0.22 dev calice072b3a33a scope link 
10.52.0.23 dev cali27c2c7edd3b scope link 
10.52.0.25 dev calie10576439bd scope link 
10.52.0.27 dev calic462804b4ce scope link 
10.52.0.28 dev cali2f59464dba9 scope link 
10.52.0.29 dev calif838a0a056b scope link 
10.52.0.33 dev calia0bf1ddd4ae scope link 
10.52.0.34 dev cali8f2636e169e scope link 
10.52.0.35 dev cali4f16a44d397 scope link 
10.52.0.36 dev calidb0ccbd54c8 scope link 
10.52.0.37 dev cali44ded6f98f6 scope link 
10.52.0.38 dev cali473471a8a05 scope link 
10.52.0.39 dev caliaf6d9627d86 scope link 
10.52.0.40 dev calic0b20777b81 scope link 
10.52.0.41 dev califa271f8858c scope link 
10.52.0.42 dev cali315006a2416 scope link 
10.52.0.43 dev cali9f6f611c8e4 scope link 
10.52.0.44 dev cali2e522e3aef6 scope link 
10.52.0.45 dev calie3b1ed09876 scope link 
10.52.0.46 dev calid5b9a994784 scope link 
10.52.0.47 dev cali74465bea224 scope link 
10.52.0.48 dev cali043c863a120 scope link 
10.52.0.50 dev cali33b7b3f862a scope link 
10.52.0.51 dev calibf3eba31e24 scope link 
10.52.0.52 dev cali61aa616e9ba scope link 
10.52.0.53 dev cali77f59883bd9 scope link 
10.52.0.54 dev caliabfb598b119 scope link 
10.52.0.55 dev cali5c966ab4f98 scope link 
10.52.0.56 dev cali52c18315662 scope link 
10.52.0.58 dev calibb04720ea74 scope link 
10.52.0.59 dev cali35e4437bf1c scope link 
10.52.0.60 dev caliec8f5b6fbfb scope link 
10.52.0.61 dev cali01d902f20cf scope link 
10.52.0.62 dev cali9807bd9c6fc scope link 
10.52.0.63 dev caliec6ac667497 scope link 
10.52.0.64 dev calif389b67fbe3 scope link 
10.52.0.65 dev caliaf5c92ab815 scope link 
10.52.0.66 dev cali0de66914e54 scope link 
10.52.0.67 dev calif29c7003ca9 scope link 
10.52.0.68 dev calid03da59fcf8 scope link 
10.52.0.69 dev cali53999b8fec4 scope link 
10.52.0.70 dev cali227c3a1ee4d scope link 
10.52.0.71 dev caliaac75846333 scope link 
10.52.0.72 dev cali0d4fc84d736 scope link 
10.52.0.73 dev calic8f14829c5d scope link 
10.52.0.77 dev cali25fa5f45aca scope link 
10.52.0.78 dev cali7c57a494840 scope link 
10.52.0.79 dev calif83974f8c40 scope link 
10.52.0.80 dev cali389fc021f1a scope link 
10.52.0.81 dev cali085daf50421 scope link 
10.52.0.82 dev calicd271a39193 scope link 
10.52.0.83 dev calia1ce1860563 scope link 
10.52.0.85 dev califb0eb90ddb6 scope link 
10.52.1.0/24 via 10.52.1.0 dev flannel.1 onlink 
10.52.2.0/24 via 10.52.2.0 dev flannel.1 onlink 
10.52.3.0/24 via 10.52.3.0 dev flannel.1 onlink
```

We found it looks like the route to mgmt-br have been removed at that time
```
harvester-268pz:/home/rancher # ip a show dev mgmt-br
4: mgmt-br: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 8c:dc:d4:0f:1b:10 brd ff:ff:ff:ff:ff:ff
    inet 10.84.132.8/22 brd 10.84.135.255 scope global mgmt-br
       valid_lft forever preferred_lft forever
    inet 10.84.132.32/32 scope global mgmt-br
       valid_lft forever preferred_lft forever
```

**Comment by TachunLin** [comment]
After we recall the reproduce steps, the reason that cause we lost all network connection may related to we set the 
inappropriate CIDR with gateway in VM network manual route like the following

```
CIDR: 10.84.132.0/0 
Gateway: 10.84.135.254
```

And we try to delete that VM network, which cause deleting the default route

This should more closely related to a vulnerability issue that would probably cause the lost of network system connection

**Comment by TachunLin** [comment]
Verified **fixed** on `v1.1.2-rc3`. Close this issue.

### Result
When we trying to add or update VM network CIDR with subnet `0` like `10.84.132.0/0`
It will display error message to prevent this action 

  ![image](https://user-images.githubusercontent.com/29251855/224634928-7974c6a8-8787-418b-8e6f-8eca902a60d2.png)
  
  ![image](https://user-images.githubusercontent.com/29251855/224635025-bde8c7a8-1640-4082-a430-496c8edeaa57.png)

### Test Information
* Test Environment: 4 nodes Harvester bare machines
* Harvester version: `v1.1.2-rc3` 

### Verify Steps
1. Create cluster network `cn1`
2. Create a vlanconfig `nc1` on cluster network `cn1` , set uplink nic to `ens1f1`
3. Create a vm network `vlan1215`
4. Set manual route with CIDR `10.84.132.0/0` and gateway `10.84.132.254` and save
5. Check the message
6. Create the vlan network `vlan1215` with the correct CIDR and gateway
7. Update the vlan network with invalid CIDR `10.84.132.1/0` and save
8. Check the message
